Std/Data/BitVec/Basic.lean

```diff
@@ -316,9 +310,18 @@ Bitwise NOT for bit vectors.
 ```
 SMT-Lib name: `bvnot`.
 -/
-protected def not (x : BitVec n) : BitVec n := -(x + .ofNat n 1)
+protected def not (x : BitVec n) : BitVec n :=
+  .ofFin ⟨(1 <<< n).pred,
+    n.one_shiftLeft.symm ▸
+      (Nat.pred_lt <| Nat.ne_of_gt <| Nat.pos_pow_of_pos _ <| Nat.zero_lt_succ _)⟩ ^^^ x
 instance : Complement (BitVec w) := ⟨.not⟩
 
+/-- The `BitVec` with value `(2^n + (i mod 2^n)) mod 2^n`.  -/
+protected def ofInt (n : Nat) (i : Int) : BitVec n :=
+  match i with
+  | Int.ofNat a => .ofNat n a
+  | Int.negSucc a => ~~~.ofNat n a
```

Comment:
It's not immediately obvious to me that this new definition is indeed faster.
The old version has some extraneous modulus, but could be optimized to `.ofFin \<2^n - 1 - a, _\>`, since `2^n - 1 - a < 2^n` holds for all `n` and all `a`, this is is two subtractions.
Your version is one bitshift, one subtraction (by one) and one xor.

Do we have a good story for micro-benchmarks in Lean? I'd love to have some clear numbers on the relative speeds here